### PR TITLE
LPD-17674 Functional Automation - Show mapped hierarchical object fields in fragment

### PR DESCRIPTION
**Pull request by chestofwonders**
Task: https://liferay.atlassian.net/browse/LPD-17674

This PR covers:

- Add fields to a given Data Set View, including a parent field and children fields from the Fields tree view.
- Make sure the fields appears in a fragment added to a page that shows the Data Set View previously created.

### CODE REVIEW

**Comment by chestofwonders** [comment]
ci:test:playwright

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/FieldsPage.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	async saveAddFieldsModal() {
+		await this.addFieldsDialog.saveButton.click();
+	}
+
+	async addParentField(field: string) {
+		await this.addFieldsDialog.fields
+			.getByText(field, {exact: true})
+			.click();
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/FieldsPage.ts:57]: Why parent needs `click()` and child needs `check()`? is it intended?

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts [hunk 2]
```diff
@@ -60,20 +60,23 @@ export class DataSetsPage {
 		await this.newDataSetModal.nameInput.fill('Data Set Sample');
 		await this.newDataSetModal.restApplicationField.click();
 		await this.newDataSetModal.restApplicationOptions
-			.getByRole('option', {name: '/data-set-manager/fields'})
+			.getByRole('option', {name: '/headless-delivery/v1.0'})
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts:63]: Soon, we'll need to parameterize this in order to create specific data sets to test separate scenarios. Ideally, we shall base on well-known API endpoints/schemas, for instance, the ones representing DSM entity persistence. As long as they reflect the complexity we need, this looks a sensible choice.

We can move on with this though

Comment (juanjofgliferay) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts:63]: DataSet parametrization will be available with the actions [PR](https://github.com/liferay-frontend/liferay-portal/pull/3978/files#diff-79653e196ebf0913d3d203ed9cec87f1aaa39fce6e1ad5a82e5f495142d8e6dbR72)

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			await applicationsMenuPage.goToHome();
+
+			await expect(
+				page.locator('.data-set-wrapper').first()
+			).toBeVisible();
+		});
+
+		await test.step('Fields are present in fragment table heading', async () => {
+			await expect(
+				page.getByRole('button', {name: 'creator.*'})
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:138]: A potential improvement for next iterations is to model the notion of "A page with the DS fragment in it" / "A page with the FDS in it" as a dedicated POM. This way, all interactions and expectations we code over the FDS would be in a single place, where some useful wrappers (like `assertFieldIsPresent(fieldName)` in this particular example) would be used by tests conveniently

Again not for this PR but just a consideration

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/FieldsPage.ts (hunk 1 above)
Comment (chestofwonders) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/FieldsPage.ts:57]: Comes for the sake of accuracy on the action. Check parents don't make them open their children tree, they need to be clicked to both get checked and make children visible, while for the children we want it to be only checked. However, both click and check would work on children.

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts (hunk 2 above)
Comment (chestofwonders) [reply modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts:63]: Sure, but we must take into account that _all_ data used in the whole chain of cases must be related. That said, the endpoint we use here comes with different fields we must take care about in subsequent places (may be a good idea to create a big and consistent bundle of test data).

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts (hunk 3 above)
Comment (chestofwonders) [reply modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:138]: I'd love this approach!

**Comment by chestofwonders** [comment]
ci:test:playwright

File: modules/test/playwright/pages/product-navigation-applications-menu/ApplicationsMenuPage.ts [hunk 4]
```diff
@@ -109,6 +109,10 @@ export class ApplicationsMenuPage {
 		await this.apiBuilderMenuItem.click();
 	}
 
+	async goToHome() {
```
Comment (juanjofgliferay) [inline modules/test/playwright/pages/product-navigation-applications-menu/ApplicationsMenuPage.ts:112]: I thin that we don't need this anymore

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+import {fdsFragmentPageTest} from './fixtures/fdsFragmentPageTest';
+import {featureFlagsTest} from '../../fixtures/featureFlagsTest';
+import {fieldsPageTest} from './fixtures/fieldsPageTest';
+import {viewsPageTest} from './fixtures/viewsPageTest';
+
+export const test = mergeTests(
+	applicationsMenuPageTest,
+	dataSetsPageTest,
+	fdsFragmentPageTest,
+	featureFlagsTest({
```
Comment (juanjofgliferay) [inline modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:20]: SF will complain for the order :grimacing:

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/FDSFragmentPage.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {ApiHelpers} from '../../../helpers/ApiHelpers';
+import {ApplicationsMenuPage} from '../../../pages/product-navigation-applications-menu/ApplicationsMenuPage';
```
Comment (juanjofgliferay) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/FDSFragmentPage.ts:9]: We don't need the `ApplicationsMenuPage` :sweat_smile:

**Review by juanjofgliferay — APPROVED** [review]
:rocket:

**Comment by chestofwonders** [comment]
ci:test:playwright

**Comment by chestofwonders** [comment]
ci:forward

**Pull request closed without merging**